File: WorkoutApplication.Domain/Workouts/ExecutedWorkout.cs
using System;
using System.Collections.Generic;
using System.Text;
using WorkoutApplication.Domain.BaseClasses;

namespace WorkoutApplication.Domain.Workouts
{
    public class ExecutedWorkout : Entity
    {
        public virtual Workout Workout { get; set; }
        public DateTime StartDateTime { get; set; }
        public DateTime EndDateTime { get; set; }
        public string Journal { get; set; }
        public int Rating { get; set; }
    }
}

File: WorkoutTracker.UI/workout-tracker/src/app/workouts/models/executed-workout.ts
import { Entity } from "app/shared/models/entity";
import { ExecutedExercise } from "./executed-exercise";
import { Workout } from "./workout";

export class ExecutedWorkout extends Entity {
  public workout: Workout;
  public startDateTime: Date;
  public endDateTime: Date;
  public journal: string;
  public rating: number;
  public exercises: Array<ExecutedExercise>;
}
